src/cephadm/tests/test_util_funcs.py

```diff
@@ -663,3 +663,66 @@ def test_call(caplog, monkeypatch, pyline, expected, call_kwargs, log_check):
         assert result == expected
     if callable(log_check):
         log_check(caplog)
+
+
+class TestWriteNew:
+    def test_success(self, tmp_path):
+        "Test the simple basic feature of writing a file."
+        dest = tmp_path / "foo.txt"
+        with _cephadm.write_new(dest) as fh:
+            fh.write("something\n")
+            fh.write("something else\n")
+
+        with open(dest, "r") as fh:
+            assert fh.read() == "something\nsomething else\n"
+
+    def test_write_ower_mode(self, tmp_path):
+        "Test that the owner and perms options function."
+        dest = tmp_path / "foo.txt"
+
+        # if this is test run as non-root, we can't really change ownership
+        uid = os.getuid()
+        gid = os.getgid()
```

Comment:
In general tests are run as a normal user AFIAK. That does mean this test will fail in such cases?

Reply:
> In general tests are run as a normal user AFIAK. 

Agreed.

> That does mean this test will fail in such cases?

Perhaps I don't understand the question, but the idea is to prevent failures when the test is run as nonroot by only "changing" the owner to what the current state is. it'll run the fchown function, but largely act a no-op.  Does that clarify things?